coderd/database/queries/notifications.sql

```diff
@@ -189,3 +189,28 @@ WHERE
 INSERT INTO notification_report_generator_logs (notification_template_id, last_generated_at) VALUES (@notification_template_id, @last_generated_at)
 ON CONFLICT (notification_template_id) DO UPDATE set last_generated_at = EXCLUDED.last_generated_at
 WHERE notification_report_generator_logs.notification_template_id = EXCLUDED.notification_template_id;
+
+-- name: GetWebpushSubscriptionsByUserID :many
+SELECT *
+FROM webpush_subscriptions
+WHERE user_id = @user_id::uuid;
+
+-- name: InsertWebpushSubscription :one
+INSERT INTO webpush_subscriptions (user_id, created_at, endpoint, endpoint_p256dh_key, endpoint_auth_key)
+VALUES ($1, $2, $3, $4, $5)
+RETURNING *;
+
+-- name: DeleteWebpushSubscriptions :exec
+DELETE FROM webpush_subscriptions
+WHERE id = ANY(@ids::uuid[]);
+
+-- name: DeleteWebpushSubscriptionByUserIDAndEndpoint :exec
+DELETE FROM webpush_subscriptions
+WHERE user_id = @user_id AND endpoint = @endpoint;
+
+-- name: DeleteAllWebpushSubscriptions :exec
+-- Deletes all existing webpush subscriptions.
+-- This should be called when the VAPID keypair is regenerated, as the old
+-- keypair will no longer be valid and all existing subscriptions will need to
+-- be recreated.
+TRUNCATE TABLE webpush_subscriptions;
```

Comment:
Do you use `TRUNCATE` instead of `DELETE FROM` because of transaction?

Reply:
I'm not sure how big this table is going to be, but `TRUNCATE` is always faster than `DELETE FROM`.

> TRUNCATE quickly removes all rows from a set of tables. It has the same effect as an unqualified DELETE on each table, but since it does not actually scan the tables it is faster. Furthermore, it reclaims disk space immediately, rather than requiring a subsequent VACUUM operation. This is most useful on large tables.


https://www.postgresql.org/docs/current/sql-truncate.html